### PR TITLE
Require C++17

### PR DESCRIPTION
**Pull request by cticenhour**
This PR removes entries for the C++11 standard in compiler flags and replaces it with C++17. We should be covered during compilation by the definition of CXXFLAGS in our most-recent `moose-mpich` package, but this helps ensure that the default standard is C++17 for those who do not use our packages when compiling hit, using `clang_autocomplete`, and in the `MooseSourceParser`. 

Refs #20377 and #20424

### CODE REVIEW

**Comment by cticenhour** [comment]
@loganharbour we can merge this or leave it if you want. I just did a search and found these during our earlier discussion on Slack.

**Review by loganharbour — CHANGES_REQUESTED** [review]
I'm not sure if we're ready for this. We need a libmesh update to be able to enforce c++17 first, and we also need to change the requirements in documentation.

**Comment by loganharbour** [comment]
@cticenhour wanna add a submodule update to https://github.com/libMesh/libmesh/commit/617928abf97fb44d6af0970b91c883e1428e1e8a in libmesh? And then update the docs?

**Comment by cticenhour** [comment]
Yeah, I can do that. Might not be today, as I plan to take part of the afternoon as PL, but by Monday for sure.

**Comment by loganharbour** [comment]
No worries, I'll get to it this afternoon then 👍🏼

**Review by loganharbour — CHANGES_REQUESTED** [review]
We still need to address https://github.com/idaholab/moose/blob/next/framework/doc/packages_config.yml; haven't gotten around to it

**Comment by loganharbour** [comment]
Nice. Looks like this is good to go other than https://github.com/idaholab/moose/pull/20397#pullrequestreview-892950550.

**Review by milljm — DISMISSED** [review]
👍

**Comment by roystgnr** [comment]
What all is left to be done here?  Just bumping up to `minimum_gcc: 7.5.0`, `minimum_clang: 5.0.2`, `minimum_intel: 19.0.1` in packages_config.yml?

**Comment by loganharbour** [comment]
@roystgnr nothing is left now.

**Review by loganharbour — DISMISSED** [review]
Approving changes that are not my own.

**Comment by roystgnr** [comment]
I'm pretty confident that this PR didn't cause `test:pressure_bc.inclined_edge_2d_pressure FAILED (EXODIFF)` in Valgrind xfem, but all the other tests are `MooseDocs.common.exceptions.MooseDocsException: The following command combination is unknown: 'package minimum_clang'.` and that actually looks like it might be our fault...

**Comment by cticenhour** [comment]
Yeah, Logan discovered late last night that removing the whole package extension might be premature considering that it has been useful in propagating our minimum requirements through the documentation. The other results might be the random failures we've been seeing lately.

**Comment by roystgnr** [comment]
Yeah, that's our fault.  Assuming we really want to go through with #20424 (and do we? maybe "we" manage our packages with conda, but it'd be really nice if that was a convenience not a requirement; "apt install mooseframework" would be nice to do someday and it's never going to happen if it requires conda), we need to grep through the remaining docs for `minimum_clang` etc. and replace those invocations.

**Comment by cticenhour** [comment]
We went back and forth with it yesterday; an ideal world to me would be tying the versions of our conda dependencies to the documentation in a similar way as the package extension did with the old `moose-environment` package. One problem with keeping the current extension around as it is....it affects our manual installation instructions (which may be woefully out of date at this point) and just de-couples things from our well-tested configurations.

**Comment by loganharbour** [comment]
I do plan on retaining the package extension now - it’ll just store less information (compilers, mpich, petsc, maybe libmesh). I’ll get around to that today.

**Comment by lindsayad** [comment]
Anything I can do to help here? I'd be inclined to limit that xfem test to one timestep as it's doing everything interesting in the first timestep anyways and then after that there's barely anything to solve

**Comment by loganharbour** [comment]
I'm finishing up the documentation stuff right now; should be good soon. Not sure what's going on with xfem... doesn't look deterministic

**Comment by lindsayad** [comment]
It's probably not, that's why I think we should change the test at some point (doesn't have to be here)

File: modules/doc/content/newsletter/2022_03.md [hunk 1]
```diff
@@ -7,6 +7,22 @@ for a complete description of all MOOSE changes.
...
+- Add auto-detection for PETSc external packages, notably HDF5
+- Add internal thread safety within `NumericVector` subclasses
+- Add initial support for `Poly2Tri` triangulations and refactor `TriangulatorInterface`
+- Optimize `InfFE::reinit` when not all data has been requested
+- Support the matrix-free case when computing the residual and Jacobian together
+- Add more robust `Packing` for communication of fixed-size classes win variable-size containers
+- Enable setting PETSc mat/vec types at runtime for future GPU use
+- Require C++17 support and begin using it internally
+- Add new static `TensorValue::rotation_matrix` helper functions
+
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2022_03.md:27]: LGTM! 👍🏻

**Review by loganharbour — APPROVED** [review]
Approving changes that are not my own.

**Review by milljm — APPROVED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**